tools/lint/common.c

```diff
@@ -209,6 +209,137 @@ fill_cmdline_file(struct ly_set *set, struct ly_in *in, const char *path, LYD_FO
     return rec;
 }
 
+int
+collect_features(const struct lys_module *mod, struct ly_set *set)
+{
+    struct lysp_feature *f = NULL;
+    uint32_t idx = 0;
+
+    while ((f = lysp_feature_next(f, mod->parsed, &idx))) {
+        if (ly_set_add(set, (void *)f->name, 1, NULL)) {
+            YLMSG_E("Memory allocation failed.\n");
+            ly_set_erase(set, NULL);
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+void
+print_features(const struct lys_module *mod, struct ly_set *set)
+{
+    size_t max_len;
+    uint32_t j;
+    const char *name;
+
+    /* header */
+    printf("%s:\n", mod->name);
+
+    if (set->count) {
+        /* get max len */
+        max_len = 0;
+        for (j = 0; j < set->count; ++j) {
+            name = set->objs[j];
+            if (strlen(name) > max_len) {
+                max_len = strlen(name);
+            }
+        }
+
+        /* print features */
+        for (j = 0; j < set->count; ++j) {
+            name = set->objs[j];
+            printf("\t%-*s (%s)\n", (int)max_len, name, lys_feature_value(mod, name) ? "off" : "on");
+        }
+    } else {
+        printf("\t(none)\n");
```

Comment:
Maybe it would be more appropriate:
`if (!set->count) {  printf("\t(none)\n\n"); return; }`
This would prevent the `if` immersion.